docs/ug/cli.md

```diff
@@ -107,6 +107,17 @@ The last modified dates will be in the same timezone specified with the `--timez
 
 <!-- ------------------------------------------------------------------------------------------------------ -->
 
+### `--optimize-cloning`, `-m`
+
+**`--optimize-cloning`**: Clones repositories using Git's shallow cloning functionality, which can significantly reduce the time taken to clone large repositories. However, the flag should not be used for smaller repositories. Note: this flag is incompatible with the `--last-modified-date`.
```

Comment:
You might want to explain why it should not be used for small repos. also, what's considered a 'small' repo?

Reply:
> 
> 
> You might want to explain why it should not be used for small repos.

Noted, will add a point about the overhead.

>
>
> also, what's considered a 'small' repo?

The exact threshold isn't very clear, since the reduction in size due to shallow cloning varies depending on when the commits were made relative to the time window of the shallow clone. But I think it is safe to say that shallow cloning would not be very helpful if the size of the output of the bare clone (i.e. the ".git" file) is less than 1GB.

Reply:
> The exact threshold isn't very clear, since the reduction in size due to shallow cloning varies depending on when the commits were made relative to the time window of the shallow clone. But I think it is safe to say that shallow cloning would not be very helpful if the size of the output of the bare clone (i.e. the ".git" file) is less than 1GB.

I see. You may want to mention this in the documentation as well.